### PR TITLE
Save voicenote functionality is added

### PR DESCRIPTION
**Pull request by mohammra**
With the saveVoiceNote function, now the saved notes show in the list

### CODE REVIEW

File: Voice Note/Views/BottomBarView.swift [hunk 1]
```diff
@@ -119,7 +123,35 @@ struct BottomBarView: View {
...
-    
+    func saveVoiceNote() {
+        guard let url = voiceNoteViewModel.fileUrlList.last else {
+            return
+        }
+
+        let newVoiceNote = VoiceNote(context: managedObjectContext)
+            newVoiceNote.id = UUID()
+            newVoiceNote.text = speechRecognizer.transcriptionText
+            newVoiceNote.title = "Note Title"
```
Comment (anwar-ulhaq) [inline Voice Note/Views/BottomBarView.swift]: how about first few words from transcription to use as a Title? @giaongo @TaiNguyen2407
`newVoiceNote.title = speechRecognizer.transcriptionText.prefix(12)+"..."`

File: Voice Note/Views/BottomBarView.swift [hunk 2]
```diff
@@ -119,7 +123,35 @@ struct BottomBarView: View {
...
+            newVoiceNote.fileUrl = url
+            newVoiceNote.createdAt = Date()
+            newVoiceNote.duration = 3765
```
Comment (anwar-ulhaq) [inline Voice Note/Views/BottomBarView.swift]: Can't we use getDuration(for: url) function of VoiceNoteViewModel? it is private, do not know if it breaks app functionality if we make it internal/public and use it here in BottomBarView. This function is not used any where in the project.

File: Voice Note/Views/BottomBarView.swift [hunk 3]
```diff
@@ -119,7 +123,35 @@ struct BottomBarView: View {
...
+            newVoiceNote.location = Location(context: managedObjectContext)
+            newVoiceNote.location?.latitude = 24.444
+            newVoiceNote.location?.longitude = 64.444
```
Comment (anwar-ulhaq) [inline Voice Note/Views/BottomBarView.swift]: lat lon @Tai working on, right?

File: Voice Note/Views/BottomBarView.swift [hunk 4]
```diff
@@ -119,7 +123,35 @@ struct BottomBarView: View {
...
+            newVoiceNote.weather = Weather(context: managedObjectContext)
+            newVoiceNote.weather?.temperature = Temperature(context: managedObjectContext)
+            newVoiceNote.weather?.temperature?.average = 34
+            newVoiceNote.weather?.temperature?.maximum = 44 
+            newVoiceNote.weather?.temperature?.minimum = 24
```
Comment (anwar-ulhaq) [inline Voice Note/Views/BottomBarView.swift]: Temperature is almost ready, hope it is presentable before review.

**Review by anwar-ulhaq — COMMENTED** [review]
Some comments for suggestion

File: Voice Note/Views/BottomBarView.swift (hunk 1 above)
Comment (anwar-ulhaq) [reply Voice Note/Views/BottomBarView.swift]: Result would be like
### `This is my vo...`

Comment (giaongo) [reply Voice Note/Views/BottomBarView.swift]: @anwar-ulhaq  Prefix part for title might make the UI look confusing when both  ListItem / RecordingList shares the similar text of the transcription and title on the same screen. If you would like to make a prefix for title, I suggest that we only display prefixed title in the RecordingListView without the full transcription text? Full transcription text only shows when user clicks to see the DetailView?

File: Voice Note/Views/BottomBarView.swift (hunk 2 above)
Comment (giaongo) [reply Voice Note/Views/BottomBarView.swift]: Sure, we should implement that function in this case.

**Review by giaongo — APPROVED** [review]
I will approve the PR for now

**Pull request closed without merging**